### ISSUE TITLE
libunwind fails to build on gcc-10 (or on CC="gcc -fno-common")

### ISSUE THREAD

**Issue by trofi** [libunwind/libunwind#154]
It's an upstream version of downstream bug: https://bugs.gentoo.org/706560

gcc-10 will change the default from -fcommon to fno-common: https://gcc.gnu.org/PR85678.
The error also happens if CFLAGS=-fno-common passed explicitly.

On gcc-10 (and gcc-9 -fno-common) libunwind build fails as:

```
libtool: link: gcc -shared  -fPIC -DPIC  .libs/os-linux.o mi/.libs/init.o mi/.libs/flush_cache.o mi/.libs/mempool.o mi/.libs/strerror.o x86_64/.libs/is_fpreg.o x86_64/.libs/regname.o x86_64/.libs/Los-linux.o mi/.libs/backtrace.o mi/.libs/dyn-cancel.o mi/.libs/dyn-info-list.o mi/.libs/dyn-register.o mi/.libs/Ldyn-extract.o mi/.libs/Lfind_dynamic_proc_info.o mi/.libs/Lget_accessors.o mi/.libs/Lget_proc_info_by_ip.o mi/.libs/Lget_proc_name.o mi/.libs/Lput_dynamic_unwind_info.o mi/.libs/Ldestroy_addr_space.o mi/.libs/Lget_reg.o mi/.libs/Lset_reg.o mi/.libs/Lget_fpreg.o mi/.libs/Lset_fpreg.o mi/.libs/Lset_caching_policy.o mi/.libs/Lset_cache_size.o x86_64/.libs/setcontext.o x86_64/.libs/Lapply_reg_state.o x86_64/.libs/Lreg_states_iterate.o x86_64/.libs/Lcreate_addr_space.o x86_64/.libs/Lget_save_loc.o x86_64/.libs/Lglobal.o x86_64/.libs/Linit.o x86_64/.libs/Linit_local.o x86_64/.libs/Linit_remote.o x86_64/.libs/Lget_proc_info.o x86_64/.libs/Lregs.o x86_64/.libs/Lresume.o x86_64/.libs/Lstash_frame.o x86_64/.libs/Lstep.o x86_64/.libs/Ltrace.o x86_64/.libs/getcontext.o  -Wl,--whole-archive ./.libs/libunwind-dwarf-local.a ./.libs/libunwind-elf64.a -Wl,--no-whole-archive  -lc -lgcc_s -llzma -lz  -g -O2 -nostartfiles -nostdlib   -Wl,-soname -Wl,libunwind.so.8 -o .libs/libunwind.so.8.0.1
/usr/lib/gcc/x86_64-pc-linux-gnu/10.0.1/../../../../x86_64-pc-linux-gnu/bin/ld: x86_64/.libs/Linit.o:(.bss+0x0): multiple definition of `_U_dyn_info_list'; mi/.libs/dyn-info-list.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
```

Looks like `_U_dyn_info_list` definition site is tricky even on amd64

```
$ git grep _U_dyn_info_list | fgrep list_t
include/libunwind_i.h:extern unw_dyn_info_list_t _U_dyn_info_list;
src/mi/dyn-info-list.c:HIDDEN unw_dyn_info_list_t _U_dyn_info_list;
src/x86/Ginit.c:HIDDEN unw_dyn_info_list_t _U_dyn_info_list;

src/x86/Linit.c:#include "Ginit.c"
```

Should it perhaps be consolidated in a single place?

**Comment by trofi** [comment]
Current master branch seems to work. Thank you!

**Comment by rzr** [comment]
It think I hit this issue on libunwind-1.3.1 , is fix released in next version 1.3.2 ?
